src/platform/input/game-pads.js

```diff
@@ -74,56 +114,487 @@ const MAPS = {
+    }
+
+    /**
+     * @type {number[]} - The values from analog axes present on the GamePad. Values are between -1 and 1.
+     */
+    get axes() {
+        return this.map.axes.map(a => this.pad.axes[MAPS.INDEXES.axes[a]] || 0);
+    }
+
+    /**
+     * @type {GamePadButton[]} - The buttons present on the GamePad. Some buttons may be null.
+     */
+    get buttons() {
+        return this.map.buttons.map(b => this._buttons[MAPS.INDEXES.buttons[b]] || null);
+    }
+
+    /**
+     * Make the gamepad vibrate.
+     *
+     * @param {number} intensity - Intensity for the vibrations, must be between 0 and 1.
+     * @param {number} duration - Duration for the vibration, in miliseconds.
+     * @param {object} [options] - Options for special vibration patern.
+     * @param {string} [options.type] - Type of patern. Available types are "dual-rumble" and "vibration". Defaults to "vibration".
+     * @param {number} [options.startDelay] - Delay before the patern starts, in miliseconds. Defaults to 0.
+     * @param {number} [options.strongMagnitude] - Intensity for strong actuators, must be between 0 and 1. Defaults to intensity.
+     * @param {number} [options.weakMagnitude] - Intensity for weak actuators, must be between 0 and 1. Defaults to intensity.
+     * @returns {Promise<boolean>} Return a Promise resulting in true if the pulse was successfully completed.
+     */
+    pulse(intensity, duration, options) {
+        const actuator = this.pad.vibrationActuator;
+
+        if (actuator) {
+            if (actuator.playEffect) {
+                if (options) {
+                    return actuator.playEffect(options.type === 'dual-rumble' ? 'dual-rumble' : 'vibration', {
+                        'duration': duration,
+                        'startDelay': options.startDelay || 0,
+                        'strongMagnitude': typeof options.strongMagnitude === 'number' ? options.strongMagnitude : intensity,
+                        'weakMagnitude': typeof options.weakMagnitude === 'number' ? options.weakMagnitude : intensity
+                    });
+                }
+
+                return actuator.playEffect('vibration', {
+                    'duration': duration,
+                    'startDelay': 0,
+                    'strongMagnitude': intensity,
+                    'weakMagnitude': intensity
+                });
+            } else if (actuator.pulse) {
+                if (options && options.startDelay) {
+                    // Custom delay
+                    return new Promise(function (resolve, reject) {
+                        setTimeout(function () {
+                            actuator.pulse(intensity, duration).then(resolve).catch(reject);
+                        }, options.startDelay);
+                    });
+                }
+
+                return actuator.pulse(intensity, duration);
+            }
+        }
+
+        return new Promise((resolve) => {
+            resolve(false);
+        });
```

Comment:
```suggestion
        return Promise.resolve(false);
```

This is shorter, but the function could also be marked as `async function`. Not sure if we want the promises still, so far we stick to IE11 support. I can't test it as I'm on Linux.

Any thoughts from others who introduced Promises previously? I see `Promise` used in `src/platform/graphics/graphics-device-create.js` for example.

Reply:
@kungfooman - good question regarding the Promises, we'll have an internal discussion on this, as that's something we should really resolve.

Reply:
@mvaligursky Always good to resolve your promises.